### ISSUE TITLE
Bulk updating/deleting things by search (RQL) filter

### ISSUE THREAD

**Issue by thjaeckle** [eclipse/ditto#1419]
Let's assume you manage thousands, hundred of thousands things which shall now e.g. all get an additional attribute or e.g. set a property.
In such a case, it would be very cumbersome to
a) first find out all of the ids of the things to update
b) consume them as a stream of ids
c) for each found id, perform another API call applying the update

This would result in thousands or even hundred of thousands API interactions.
I think it would (with all of the tools Ditto has gathered) be rather easy to instead provide a bulk update API.

Example:
* a manufacturer changes its name from e.g. "Raider" to "Twix"
* you want to adjust all things with a "manufacturer" attribute being "Raider" to now instead being "Twix"

## Example HTTP API

```
PUT /api/2/bulk/things/attributes/manufacturer?filter=eq(attributes/manufacturer,'Raider')
HTTP payload: "Twix"
```

Deleting things could work in a very similar way:
```
DELETE /api/2/bulk/things?filter=eq(attributes/manufacturer,'Raider')
```

## Example Ditto Protocol

```json
{
  "topic": "_/_/things/twin/bulk/modify",
  "headers": {},
  "path": "/attributes/manufacturer",
  "value": {
    "bulkFilter": "eq(attributes/manufacturer,'Raider')"
    "bulkValue": "Twix"
  }
}
```

**Comment by Korakbhakta02** [comment]
HI @thjaeckle is the bulk updating/deleting service present currently in ditto?

**Comment by thjaeckle** [comment]
No, the issue is still open.

**Comment by thjaeckle** [comment]
@thomas-fries-keen no, Mongo transactions are not in scope, this would not be possible with the Pekko persistence plugin and event sourcing in general.  
I would even say transactions will never be in scope of Ditto.

> If I want to change a set of devices in an "all-or-nothing" manner.

I don't see how Ditto could solve that. Using the Ditto API, you can do that of course by your own. 
Eg using the history Api of a thing, you could fetch the state of a thing before a change was applied and re-apply that state again, effectively undoing the latest change.
